From 75497eb3bb57424cefbbbe0c61cd2b0adcad802b Mon Sep 17 00:00:00 2001
From: Andreas Schneider <asn@samba.org>
Date: Wed, 2 Nov 2016 17:19:09 +0100
Subject: [PATCH] s3-winbind: Do not return NO_MEMORY if we have an empty user
 list

The domain child for the MACHINE ACCOUNT might fail with
NT_STATUS_NO_MEMORY because an emtpy user list is returned.

*pnum_info is already set to 0 at the beginngin so we should just
declare victory here!

BUG: https://bugzilla.samba.org/show_bug.cgi?id=12405

Signed-off-by: Andreas Schneider <asn@samba.org>
---
 source3/winbindd/winbindd_rpc.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/source3/winbindd/winbindd_rpc.c b/source3/winbindd/winbindd_rpc.c
index c560a6b..59bd78a 100644
--- a/source3/winbindd/winbindd_rpc.c
+++ b/source3/winbindd/winbindd_rpc.c
@@ -88,6 +88,10 @@ NTSTATUS rpc_query_user_list(TALLOC_CTX *mem_ctx,
 		num_dom_users = disp_info.info1.count;
 
 		num_info += num_dom_users;
+		/* If there are no user to enumerate we're done */
+		if (num_info == 0) {
+			return NT_STATUS_OK;
+		}
 
 		info = TALLOC_REALLOC_ARRAY(mem_ctx,
 					    info,
-- 
2.7.4

